account_payment_purchase/models/account_invoice.py

```diff
@@ -10,6 +10,9 @@ class AccountMove(models.Model):
 
     @api.onchange("purchase_vendor_bill_id", "purchase_id")
     def _onchange_purchase_auto_complete(self):
+        if not self.purchase_id:
+            # User did not add a purchase order, no need to warn
+            return super()._onchange_purchase_auto_complete()
```

Comment:
```suggestion
            return super()._onchange_purchase_auto_complete() or {}
```
🤔

Reply:
No that was needed in this method for adding the warning, but returning directly is not needed, whatever the return is.